packages/pigeon/README.md

```diff
@@ -1,13 +1,354 @@
 # Pigeon
 
+// TODO: how to frame this?
+// To get started, see the Quickstart Guide in the [Example README](./example/README.md).
+
 Pigeon is a code generator tool to make communication between Flutter and the
 host platform type-safe, easier, and faster.
 
-Pigeon removes the necessity to manage strings across multiple platforms and languages.
-It also improves efficiency over common method channel patterns. Most importantly though,
-it removes the need to write custom platform channel code, since pigeon generates it for you.
+Pigeon works by reading special files which by convention are placed in a separate
+folder, `/pigeons`, alongside your `/lib` folder. You can put all of your definitions
```

Comment:
This is a flutter/packages repo convention; there's no reason for it to be a convention anywhere else (or for there to even be a convention in general outside of our repo).

Reply:
Is it fair to say that these files should be outside of the `/lib` folder?

Reply:
I think it should be outside the `/lib` folder, yes. It isn't code that is actively used by anything other than during pigeon generation.